pkg/api/persistentvolume/util_test.go

```diff
@@ -71,8 +71,8 @@ func TestPVSecrets(t *testing.T) {
 		{Spec: api.PersistentVolumeSpec{
 			ClaimRef: &api.ObjectReference{Namespace: "claimrefns", Name: "claimrefname"},
 			PersistentVolumeSource: api.PersistentVolumeSource{
-				ScaleIO: &api.ScaleIOVolumeSource{
-					SecretRef: &api.LocalObjectReference{
+				ScaleIO: &api.ScaleIOPersistentVolumeSource{
+					SecretRef: &api.SecretReference{
 						Name: "Spec.PersistentVolumeSource.ScaleIO.SecretRef"}}}}},
```

Comment:
add a second PV definition that exercises the version that specifies a namespace:
```
 		{Spec: api.PersistentVolumeSpec{
 			ClaimRef: &api.ObjectReference{Namespace: "claimrefns", Name: "claimrefname"},
 			PersistentVolumeSource: api.PersistentVolumeSource{
				ScaleIO: &api.ScaleIOPersistentVolumeSource{
					SecretRef: &api.SecretReference{
 						Name: "Spec.PersistentVolumeSource.ScaleIO.SecretRef"}}}}},
 		{Spec: api.PersistentVolumeSpec{
 			ClaimRef: &api.ObjectReference{Namespace: "claimrefns", Name: "claimrefname"},
 			PersistentVolumeSource: api.PersistentVolumeSource{
				ScaleIO: &api.ScaleIOPersistentVolumeSource{
					SecretRef: &api.SecretReference{
 						Name: "Spec.PersistentVolumeSource.ScaleIO.SecretRef",
 						Namespace: "scaleio-ns"}}}}},
```

and add to `expectedNamespacedNames`:
```
"scaleio-ns/Spec.PersistentVolumeSource.ScaleIO.SecretRef"
```

Reply:
that will reveal that `util.VisitPVSecretNames` needs a corresponding change to use the namespace from the secretRef if specified, otherwise using the namespace from the bound PVC for backwards compatibility (see CephFS as an example)